packages/itwin/grouping-mapping-widget/src/widget/components/QueryBuilder.ts

```diff
@@ -45,9 +61,6 @@ export class QueryBuilder {
   public dataProvider: PresentationPropertyDataProvider | undefined;
   public query: Query | undefined;
```

Comment:
Where is the `query` property used outside of `QueryBuilder`? The only public members should be `addProperty`, `removeProperty` and `buildQuery`.

Reply:
Changed all properties to private.

`query` property was used in `GroupQueryBuilderContainer.tsx` to reset `query` to `undefined`, added `public resetQuery` method instead.
`dataProvider` property was used in `PropertyGridWrapper.tsx` to set `dataProvider`, added `public setDataProvider` method instead.

Since `QueryBuilder` constructor is called with `undefined` argument (sets initial `dataProvider` value) everywhere except for tests, should I remove it from constructor altogether?

Reply:
I don't think the `QueryBuilder` can function without a `dataProvider`. I think it should be made mandatory in ctor.

Reply:
I don't think that's possible, `QueryBuilder` is created in `GroupAction.tsx` initially, however as that point we don't have `dataProvider` yet

Reply:
Can the `QueryBuilder` do anything without `dataProvider`?

Reply:
Made `dataProvider` required in `QueryBuilder` ctor